lib/services/services_linux.c

```diff
@@ -679,11 +679,14 @@ async_action_complete(mainloop_child_t *p, pid_t pid, int core, int signo,
     } else if (mainloop_child_timeout(p)) {
         const char *reason = NULL;
 
-        if (op->rsc != NULL) {
-            reason = "Resource agent did not complete in time";
-        } else if (pcmk__str_eq(op->standard, PCMK_RESOURCE_CLASS_STONITH,
-                                pcmk__str_none)) {
+        if (pcmk__str_eq(op->standard, PCMK_RESOURCE_CLASS_STONITH,
+                         pcmk__str_none)) {
             reason = "Fence agent did not complete in time";
+        } else if (pcmk__str_eq(op->standard, PCMK_RESOURCE_CLASS_ALERT,
+                                pcmk__str_none)) {
+            reason = "Alert agent did not complete in time";
+        } else if (op->standard != NULL) {
+            reason = "Resource agent did not complete in time";
```

Comment:
Probably not a good example for a generic string-to-string mapping ... especially not using hash-maps, trees or things alike ...

Reply:
Eventually, I want to make the agent standards into objects. That would be a good time to add a member or method for what to call it. I'm tempted to do it every time I work with the standards code ...

Reply:
like with e.g. an enum as member to be used instead of string-comparison?

Reply:
I'm thinking there would be an array or hash table with an entry for each supported standard. If it's an array, then an enum for the array indexes would make sense, and if a hash table, then the standard name would be used to look up the object. But rather than a mapping function I imagine a member e.g. "%s did not complete in time", standard->desc.

I haven't thought it through, so it might end up being very different